boards/arm/stm32wl5/nucleo-wl55jc/Kconfig

```diff
@@ -24,547 +24,8 @@ config ARCH_BOARD_ENABLE_CPU2
-	bool "Not available"
-
-config SSD1680_GPIO_DTA_CMD_PORTA
-	bool "Port A"
-
-config SSD1680_GPIO_DTA_CMD_PORTB
-	bool "Port B"
-
-config SSD1680_GPIO_DTA_CMD_PORTC
-	bool "Port C"
-
-config SSD1680_GPIO_DTA_CMD_PORTD
-	bool "Port D"
-
-endchoice
-
-if SSD1680_GPIO_DTA_CMD_PORTA || SSD1680_GPIO_DTA_CMD_PORTB || SSD1680_GPIO_DTA_CMD_PORTC || SSD1680_GPIO_DTA_CMD_PORTD
-config SSD1680_GPIO_PIN_DTA_CMD
-	int "Pin that switch between command and data on 4-wire SPI bus"
-	default "9" if ARCH_BOARD_NUCLEO_WL55JC
-	range 0 15
-endif #SSD1680_GPIO_DTA_CMD_PORTA || SSD1680_GPIO_DTA_CMD_PORTB || SSD1680_GPIO_DTA_CMD_PORTC || SSD1680_GPIO_DTA_CMD_PORTD
-
-choice
-	prompt "Port that handles reset line (output)"
-	default SSD1680_GPIO_RST_PORTC if ARCH_BOARD_NUCLEO_WL55JC
-
-config SSD1680_GPIO_RST_DISABLED
-	bool "Not available"
-
-config SSD1680_GPIO_RST_PORTA
-	bool "Port A"
-
-config SSD1680_GPIO_RST_PORTB
-	bool "Port B"
-
-config SSD1680_GPIO_RST_PORTC
-	bool "Port C"
-
-config SSD1680_GPIO_RST_PORTD
-	bool "Port D"
-endchoice
-
-if SSD1680_GPIO_RST_PORTA || SSD1680_GPIO_RST_PORTB || SSD1680_GPIO_RST_PORTC || SSD1680_GPIO_RST_PORTD
-config SSD1680_GPIO_PIN_RST
-	int "Pin that handles the reset line (output)"
-	default "2" if ARCH_BOARD_NUCLEO_WL55JC
-	range 0 15
-endif #SSD1680_GPIO_RST_PORTA || SSD1680_GPIO_RST_PORTB || SSD1680_GPIO_RST_PORTC || SSD1680_GPIO_RST_PORTD
-
-choice
-	prompt "Port that handles busy line (input)"
-	default SSD1680_GPIO_BUSY_PORTC if ARCH_BOARD_NUCLEO_WL55JC
-
-config SSD1680_GPIO_BUSY_DISABLED
-	bool "Not available"
-
-config SSD1680_GPIO_BUSY_PORTA
-	bool "Port A"
-
-config SSD1680_GPIO_BUSY_PORTB
-	bool "Port B"
-
-config SSD1680_GPIO_BUSY_PORTC
-	bool "Port C"
-
-config SSD1680_GPIO_BUSY_PORTD
-	bool "Port D"
-
-endchoice
-
-if SSD1680_GPIO_BUSY_PORTA || SSD1680_GPIO_BUSY_PORTB || SSD1680_GPIO_BUSY_PORTC || SSD1680_GPIO_BUSY_PORTD
-config SSD1680_GPIO_PIN_BUSY
-	int "Pin that handles the busy line (input)"
-	default "1" if ARCH_BOARD_NUCLEO_WL55JC
-	range 0 15
-endif #SSD1680_GPIO_BUSY_PORTA || SSD1680_GPIO_BUSY_PORTB || SSD1680_GPIO_BUSY_PORTC || SSD1680_GPIO_BUSY_PORTD
-
-choice
-	prompt "Port that handles power line"
-	default SSD1680_GPIO_PWR_DISABLED if ARCH_BOARD_NUCLEO_WL55JC
-
-config SSD1680_GPIO_PWR_DISABLED
-	bool "Not available"
-
-config SSD1680_GPIO_PWR_PORTA
-	bool "Port A"
-
-config SSD1680_GPIO_PWR_PORTB
-	bool "Port B"
-
-config SSD1680_GPIO_PWR_PORTC
-	bool "Port C"
-
-config SSD1680_GPIO_PWR_PORTD
-	bool "Port D"
-
-endchoice
-
-if SSD1680_GPIO_PWR_PORTA || SSD1680_GPIO_PWR_PORTB || SSD1680_GPIO_PWR_PORTC || SSD1680_GPIO_PWR_PORTD
-config SSD1680_GPIO_PIN_PWR
-	int "Pin that handles the pwr on/off line (output)"
-	range 0 31
-endif #SSD1680_GPIO_PWR_PORTA || SSD1680_GPIO_PWR_PORTB || SSD1680_GPIO_PWR_PORTC || SSD1680_GPIO_PWR_PORTD
-
-
-config SSD1680_SPI_BUS
-	int "Spi bus no"
-	range 1 2
-	default "1"
-endmenu
-
-endif #LCD_SSD1680
+source "boards/arm/stm32wl5/nucleo-wl55jc/kconfig.d/ipcc.kconfig"
```

Comment:
why create kconfig.d folder?

Reply:
How else would you split it then? kconfig.d seemed like best directory to keep all parts of kconfig.

Reply:
I saw other project split Kconfig to small part, but all these files belong to the same directory of Makefile or source file.
For examples, https://github.com/torvalds/linux/tree/master/libKconfig:
```
Kconfig
Kconfig.debug
Kconfig.kasan
Kconfig.kcsan
Kconfig.kfence
Kconfig.kgdb
Kconfig.ubsan
```
I would prefer to follow Linux practice that:

1. Name the splitted config as Kconfig.xxx
2. Don't create the new directory for Kconfig

Reply:
I agree with Kconfig.xxx. I agree to "compress" the splits a little bit.

I don't agree with not creating directory. Going with this logic then, why do we create hardware directory? We could simply name files like stm32wl5_hardware_rcc.h / stm32wl5_registers_rc.h. There are a lot of peripherals, some have more complicated configurations - like serial can also have configurations for 1wire and rs485 for each port (I did not include them in stm32wl5 as I did not implement nor tested them) which probably will be added in the future. At the end I suspect there might be event 10-15 kconfig files, so I think - just like with hardware directory - it's best to create kconfig.d.

But then again, I am not religious or anything about creating kconfig.d. It's just my point of view and proposition, if I get outvoted then I will change this, no problem.